optuna/samplers/_gp/sampler.py

```diff
@@ -214,42 +223,77 @@ def sample_relative(
+        )
 
-        if self._kernel_params_cache is not None and len(
-            self._kernel_params_cache.inverse_squared_lengthscales
+        if self._kernel_params_cache_list is not None and len(
+            self._kernel_params_cache_list[0].inverse_squared_lengthscales
         ) != len(internal_search_space.scale_types):
             # Clear cache if the search space changes.
-            self._kernel_params_cache = None
-
-        kernel_params = gp.fit_kernel_params(
-            X=normalized_params,
-            Y=standardized_score_vals,
-            is_categorical=(
-                internal_search_space.scale_types == gp_search_space.ScaleType.CATEGORICAL
-            ),
-            log_prior=self._log_prior,
-            minimum_noise=self._minimum_noise,
-            initial_kernel_params=self._kernel_params_cache,
-            deterministic_objective=self._deterministic,
-        )
-        self._kernel_params_cache = kernel_params
+            self._kernel_params_cache_list = None
+
+        kernel_params_list = []
+        n_objectives = standardized_score_vals.shape[-1]
+        is_categorical = internal_search_space.scale_types == gp_search_space.ScaleType.CATEGORICAL
+        for i in range(n_objectives):
+            cache = self._kernel_params_cache_list and self._kernel_params_cache_list[i]
+            assert isinstance(cache, gp.KernelParamsTensor) or cache is None
+            kernel_params_list.append(
+                gp.fit_kernel_params(
+                    X=normalized_params,
+                    Y=standardized_score_vals[:, i],
+                    is_categorical=is_categorical,
+                    log_prior=self._log_prior,
+                    minimum_noise=self._minimum_noise,
+                    initial_kernel_params=cache,
+                    deterministic_objective=self._deterministic,
+                )
+            )
+        self._kernel_params_cache_list = kernel_params_list
 
         best_params: np.ndarray | None
         if self._constraints_func is None:
-            acqf_params = acqf.create_acqf_params(
-                acqf_type=acqf.AcquisitionFunctionType.LOG_EI,
-                kernel_params=kernel_params,
-                search_space=internal_search_space,
-                X=normalized_params,
-                Y=standardized_score_vals,
-            )
-            best_params = normalized_params[np.argmax(standardized_score_vals), :]
+            if n_objectives == 1:
+                assert len(kernel_params_list) == 1
+                acqf_params = acqf.create_acqf_params(
+                    acqf_type=acqf.AcquisitionFunctionType.LOG_EI,
+                    kernel_params=kernel_params_list[0],
+                    search_space=internal_search_space,
+                    X=normalized_params,
+                    Y=standardized_score_vals[:, 0],
+                )
+                best_params = normalized_params[np.argmax(standardized_score_vals), np.newaxis]
```

Comment:
The shape of `best_params` is changed. Is `self._optimize_acqf` designed to work with the new shape?

Reply:
Yes, it is.

![image](https://github.com/user-attachments/assets/9e92b27d-15d7-4993-8174-eeb48e908537)